public/js/user/helpOthers.js

```diff
@@ -0,0 +1,53 @@
+const nextBtn = document.querySelector('.next');
+const doneBtn = document.querySelector('.done');
+const personalForm = document.querySelector('.personal .form');
+const requestForm = document.querySelector('.request .form');
+const personalSection = document.querySelector('.personal');
+const requestSection = document.querySelector('.request');
+const personalInfo = {};
+
+// next button event listener
+nextBtn.addEventListener('click', (e) => {
+  // checks for the error message and removes it
+  const personalFormError = document.querySelector('.personal-err');
+  if (personalFormError) personalForm.removeChild(personalFormError);
+
+  // collect the data from the form
+  const formData = new FormData(personalForm);
+  formData.forEach((value, key) => {
+    personalInfo[key] = value;
+  });
+  const { fullName, phoneNumber, email } = personalInfo;
+
+  // validate the fields
+  if (!fullName) {
+    const errorMsg = document.createElement('span');
+    errorMsg.appendChild(document.createTextNode('Please enter your full name'));
+    errorMsg.classList = 'personal-err error';
+    personalForm.insertBefore(errorMsg, nextBtn);
+  } else if (!phoneNumber) {
+    const errorMsg = document.createElement('span');
+    errorMsg.appendChild(document.createTextNode('Please enter your phone number'));
+    errorMsg.classList = 'personal-err error';
+    personalForm.insertBefore(errorMsg, nextBtn);
+  } else if (!/^[0-9]{10}$/.test(phoneNumber)) {
+    const errorMsg = document.createElement('span');
+    errorMsg.appendChild(document.createTextNode('Please enter a valid phone number'));
+    errorMsg.classList = 'personal-err error';
+    personalForm.insertBefore(errorMsg, nextBtn);
+  } else if (!email) {
+    const errorMsg = document.createElement('span');
+    errorMsg.appendChild(document.createTextNode('Please enter your email'));
+    errorMsg.classList = 'personal-err error';
+    personalForm.insertBefore(errorMsg, nextBtn);
+  } else if (!(email.includes('@') && email.includes('.'))) {
```

Comment:
@ahmedisam99 Hey, I noticed that @aminalakhsham wrote an email regex in another PR (checkout https://github.com/FACG6/enhance-a-smile/pull/98/files#r267353122), so you two could work together to reuse that here by pulling it out into another file like you all did with querySelectors.

Reply:
:+1: